src/test/java/dk/kb/netarchivesuite/solrwayback/SolrStatsTest.java

```diff
@@ -0,0 +1,147 @@
+        String[] links = new String[]{"http://kb.dk/", "http://kb.dk/en"};
+
+        ArrayList<String> crawlTimes = new ArrayList<String>();
+        crawlTimes.add("2018-03-15T12:31:51Z");
+        crawlTimes.add("2018-03-15T12:34:37Z");
+        crawlTimes.add("2018-03-15T12:35:56Z");
+        crawlTimes.add("2018-03-15T12:36:14Z");
+        crawlTimes.add("2018-03-15T12:36:43Z");
+        crawlTimes.add("2018-03-15T12:37:32Z");
+        crawlTimes.add("2018-03-15T12:37:52Z");
+        crawlTimes.add("2018-03-15T12:39:15Z");
+        crawlTimes.add("2018-03-15T12:40:09Z");
+
+        int i =1;
+        for (String crawl : crawlTimes){
+            SolrInputDocument document = new SolrInputDocument();
+            String id = ""+i++;
+            String title = "title "+i;
+            document.addField("source_file_offset", i+"");
+            document.addField("id", id);
+            document.addField("title", title);
+            document.addField("url", url);
+            document.addField("url_norm", url);
+            document.addField("record_type","response");
+            document.addField("source_file_path", "some.warc");
+            document.setField("crawl_date", crawl);
+
+            document.addField("status_code", "200");
+            document.addField("content_length", i);
+            document.addField("crawl_year", 2003);
+            document.addField("content_text_length", i);
+            document.addField("image_height", 800);
+            document.addField("image_width", 600);
+            document.addField("image_size", 480000);
+            document.addField("wayback_date", "20230417100000");
+            document.addField("links", links);
+            document.addField("domain", "www.kb.dk");
+
+            embeddedServer.add(document);
+
+        }
+        embeddedServer.commit();
+    }
+
+    /**
+     * @throws java.lang.Exception
+     */
+    @After
+    public void tearDown() throws Exception {
+        coreContainer.shutdown();
+        embeddedServer.close();
+    }
+
+    @Test
+    public void singleNumericFieldStatTest(){
+        // Testing with hardcoded documents above
+        String stats = SolrStats.getStatsForSingleNumericField("*:*", "crawl_year");
+
+        JsonObject entryAsJsonObject = extractJsonObjectFromJsonArrayString(stats);
+
+        Assert.assertEquals("crawl_year", entryAsJsonObject.get("name").getAsString());
+        Assert.assertEquals(2003.0, entryAsJsonObject.get("min").getAsDouble(), 0);
+        Assert.assertEquals(2003.0, entryAsJsonObject.get("max").getAsDouble(), 0);
+        Assert.assertEquals(18027.0, entryAsJsonObject.get("sum").getAsDouble(), 0);
+        Assert.assertEquals(9, entryAsJsonObject.get("count").getAsInt());
+        Assert.assertEquals(0, entryAsJsonObject.get("missing").getAsInt());
+        Assert.assertEquals(2003.0, entryAsJsonObject.get("mean").getAsDouble(), 0);
+    }
+
+    @Test
+    public void multipleNumericFieldStatsTest(){
```

Comment:
Maybe check that there is a sub-JSON for each requested field? (Just existende - no need to check all details for all fields)